### PR TITLE
Add gRPC client and server

### PR DESCRIPTION
**Pull request by dustinspecker**
**Note**: This pull request targets the library branch instead of main so that the diff is easier to read.

- add protobuf for Job Worker Service
- add gRPC server/client using mTLS for auth
- users may use the client to run jobs, query status, stream output, and stop jobs
- users may not interact with a job started by another user
- add some integration tests

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+		// TODO: consider returning Not Found instead of Permission Denied to hide job existence
+		return ErrUserUnauthorized
+	}
+
+	jobOutput := job.job.Stream()
+	s.mutex.Unlock()
+
+	for {
+		buffer := make([]byte, 1024)
```
Comment (sclevine) [inline internal/server/server.go:151]: Is it necessary to create the buffer on each iteration?

File: internal/server/server.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to get user from certificate: %w", err)
+	}
+
+	if user != job.user {
+		// TODO: consider returning Not Found instead of Permission Denied to hide job existence
+		return nil, ErrUserUnauthorized
+	}
+
+	if err := job.job.Stop(); err != nil {
```
Comment (sclevine) [inline internal/server/server.go:202]: Does the global mutex need to be locked for this?

File: internal/server/server.go (hunk 1 above)
Comment (dustinspecker) [reply internal/server/server.go]: Thank you for catching that - it is not.

File: internal/server/server.go (hunk 2 above)
Comment (dustinspecker) [reply internal/server/server.go]: Thank you for catching this one too. Updated code to release mutex before invoking stop.

**Comment by dustinspecker** [comment]
I force pushed a rebase. You can view the compare above to show no diff is present: https://github.com/dustinspecker/linux-job-worker-service/compare/f94166bb4976ec4786485f5186de705370d1bf67..bbdc93c031dbbbbaf78228938c9f5ca574971e1f

**Pull request merged**